File: calculus/vectors/javascript/cross-product.js
/**
 * Curried cross-product function.
 * Accepts the first 3-position array and returns a function which accepts the 
 * second 3-position array, which returns the cross product of the two. Note that
 * the order in which the arrays are passed in matters.
 * 
 */
var crossProduct = function(a) {
  var i1 = a[0],
    j1 = a[1],
    k1 = a[2];
  return function(b) {
    var i2 = b[0],
      j2 = b[1],
      k2 = b[2];
    return [j1 * k2 - k1 * j2, k1 * i2 - i1 * k2, i1 * j2 - j1 * i2];
  };
};

File: calculus/vectors/java/CrossProduct.java
public class CrossProduct {
    /**
     * Calculate the cross product of two vectors
     * in R3 represented by double arrays
     * @param v1 - first vector
     * @param v2 - second vector
     * @return the result of the cross product
     */
    public static double[] crossProduct(double[] v1, double[] v2) {
        if (v1.length != 3 || v2.length != 3) {
            throw new IllegalArgumentException("Cross product is only defined for two vectors of R3");
        }
        return new double[] {
                v1[1] * v2[2] - v1[2] * v2[1],
                v1[2] * v2[0] - v1[0] * v2[2],
                v1[0] * v2[1] - v1[1] * v2[0],
        };
    }
}